### PR TITLE
MGDOBR-265 - Integration tests: create a job executing tests against …

### PR DESCRIPTION
**Pull request by jiripetrlik**
…dev cluster

[MGDOBR-265](https://issues.redhat.com/browse/MGDOBR-265) 

As part of CI we should make sure that the OpenBridge deployed on dev cluster is up and running. Such job will make sure that dev cluster is always available, catch any error missed during PR checks and run tests against environment close to the production.

The tests can be executed periodically, possibly once in an hour.

Acceptance Criteria
Analyze best tool for running such test (GitHub actions vs Jenkins)
Implements cron job executing integration tests against dev cluster

Please make sure that your PR meets the following requirements:

- [* ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [* ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [-] All new functionality is tested
- [*] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [*] Pull Request contains link to the JIRA issue
- [*] Pull Request contains link to any dependent or related Pull Request
- [*] Pull Request contains description of the issue
- [*] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

* <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

* <b>Rebase the pull request</b>  
  Comment with: `/rebase`.

</details>

### CODE REVIEW

File: .github/workflows/dev-cluster-check.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      - name: Free disk space
+        uses: ./.github/actions/free-disk-space
+      - name: Java and Maven Setup
+        uses: ./.github/actions/java-maven-setup
+        with:
+          cache-key-prefix: ${{ runner.os }}
+      - name: Compile sandbox repo
+        uses: ./.github/actions/maven
+        with:
+          maven-command: clean verify
```
Comment (sutaakar) [inline .github/workflows/dev-cluster-check.yml:21]:
```suggestion
          maven-command: clean verify -DskipTests
```
To speed up the execution.

File: .github/workflows/dev-cluster-check.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      - name: Run integration tests
+          uses: ./.github/actions/maven
+          with:
+            maven-command: clean verify -Pcucumber -Devent-bridge.manager.url=${{ secrets.DEV_CLUSTER_MANAGER_URL }} -Dkey-cloak.url=${{ secrets.DEV_CLUSTER_KEY_CLOAK_URL }}
```
Comment (sutaakar) [inline .github/workflows/dev-cluster-check.yml:25]: You will most likely need to pass also credentials for Keycloak.

File: .github/workflows/dev-cluster-check.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+name: Check development cluster
+on:
+  schedule:
+    - cron: '* */3 * * *'
```
Comment (sutaakar) [inline .github/workflows/dev-cluster-check.yml:4]: Is this valid?
According to https://www.freeformatter.com/cron-expression-generator-quartz.html it should be something like `0 0 * ? * *`?

File: .github/workflows/dev-cluster-check.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            maven-command: clean verify -Pcucumber -Devent-bridge.manager.url=${{ secrets.DEV_CLUSTER_MANAGER_URL }} -Dkey-cloak.url=${{ secrets.DEV_CLUSTER_KEY_CLOAK_URL }} -Dbridge.token.username=${{ secrets.DEV_CLUSTER_BRIDGE_TOKEN_USERNAME }} -Dbridge.token.password=${{ secrets.DEV_CLUSTER_BRIDGE_TOKEN_PASSWORD }}
+            working-directory: integration-tests
+      - name: Publish Test Report
+          if: ${{ always() }}
+          uses: scacap/action-surefire-report@v1.0.10
+          with:
+            skip_publishing: true
+            check_name: Test Results
+            fail_on_test_failures: true
+            fail_if_no_tests: false
```
Comment (sutaakar) [inline .github/workflows/dev-cluster-check.yml:34]: Please fix the wrong alignment

**Comment by jiripetrlik** [comment]
@sutaakar 
Thank you for your review. I've updated PR according to your comments.

File: .github/workflows/dev-cluster-check.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        with:
+          cache-key-prefix: ${{ runner.os }}
+      - name: Compile sandbox repo
+        uses: ./.github/actions/maven
+        with:
+          maven-command: clean install -DskipTests
+      - name: Run integration tests
+        uses: ./.github/actions/maven
+        with:
+          maven-command: clean verify -Pcucumber -Devent-bridge.manager.url=${{ secrets.DEV_CLUSTER_MANAGER_URL }} -Dkey-cloak.url=${{ secrets.DEV_CLUSTER_KEY_CLOAK_URL }} -Dbridge.token.username=${{ secrets.DEV_CLUSTER_BRIDGE_TOKEN_USERNAME }} -Dbridge.token.password=${{ secrets.DEV_CLUSTER_BRIDGE_TOKEN_PASSWORD }}
```
Comment (r00ta) [inline .github/workflows/dev-cluster-check.yml:31]: are the tests going to cleanup all the resources from the cluster with some `delete` operations?

Comment (sutaakar) [reply .github/workflows/dev-cluster-check.yml:31]: Cleanup is part of test teardown - https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/End2EndTestITStep.java#L256-L265

File: .github/workflows/dev-cluster-check.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
+name: Check development cluster
+on:
+  schedule:
+    - cron: '* 0/2 * * *'
```
Comment (sutaakar) [inline .github/workflows/dev-cluster-check.yml:4]: thinking if the expression should be rather like:
```suggestion
    - cron: '0 0/2 * * *'
```

**Review by sutaakar — APPROVED** [review]

**Review by kostola — APPROVED** [review]
Assuming that the `maven-command` that launches Cucumber is correct (which I can't tell because I don't know how it works exactly), this PR looks good to me.

**Pull request merged**